tasks/managed/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```diff
@@ -143,6 +143,21 @@ spec:
           local identity=$1
           local reference=$2
           local digest=$3
+
+          # cosign has very limited support for selecting the right auth entry,
+          # so create a custom auth file with just one entry.
+          SOURCE_AUTH_FILE_DIR=$(mktemp -d)
+          mkdir -p "${SOURCE_AUTH_FILE_DIR}"  # the tests mock `mktemp`, so we ensure it exists
```

Comment:
Rather than this, I would suggest having this in the mock:
```
function mktemp() {
if [[ ${1:-} == -d ]]; then
  /usr/bin/mktemp -d
else
  echo "temp_key_file"
fi
}
```

Reply:
updated